silverback-integration-kafka/src/Silverback.Integration.Kafka/Messaging/Broker/Kafka.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Confluent.Kafka;
+
+namespace Silverback.Messaging.Broker
+{
+    public class Kafka : Broker
+    {
+        private readonly ConcurrentBag<(string topic,
+            Dictionary<string, object> settings,
+            Func<IDeliveryResultHandler> handlerFactory,
+            IProducer producer)> _producers = new ConcurrentBag<(string, Dictionary<string, object>, Func<IDeliveryResultHandler>, IProducer)>();
+
+        /// <inheritdoc />
+        public override IConsumer GetConsumer(IEndpoint endpoint)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc />
+        public override IProducer GetProducer(IEndpoint endpoint)
+        {
+            var producerSettings = _producers.First(p => p.topic == endpoint.Name);
+            if (producerSettings.producer != null)
+                return producerSettings.producer;
+
+            producerSettings.producer = new KafkaProducer(endpoint,
+                producerSettings.settings,
+                new DeliveryHandler(producerSettings.handlerFactory));
+
+            return producerSettings.producer;
+        }
+
+        #region configurations
+        /// <summary>
+        /// Add the producer to the broker
+        /// </summary>
+        public Kafka AddProducer(string topicName, Dictionary<string, object> settings)
```

Comment:
I'm a bit unsure about this approach.
Beside that one should be able to add a default.